web/src/echo.js

```diff
@@ -463,6 +467,7 @@ function abort_message(message) {
 export function display_slow_send_loading_spinner(message) {
     const $rows = message_lists.all_rendered_row_for_message_id(message.id);
     if (message.locally_echoed && !message.failed_request) {
+        message.show_slow_send_spinner = true;
```

Comment:
What clears this if the message is successfully delivered?

Reply:
> What clears this if the message is successfully delivered?

Nothing similar to `failed_request`. I can add a commit for that.

Reply:
I am not sure that's the best idea though since attaching it to `message` is really convinient when rending the message anywhere and we don't have to remember if we checked a `Map` `slowing_sending_message` or `failed_messages`.


Might be better pursing this in a differnet thread?

Reply:
Yeah maybe that should be a follow-up refactor -- but we should probably just have a more general `sending_state` map that contains a few fields associated with the message that we can work with -- or add them to the main `Message` type -- the latter might be most reasonable actually.